core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -679,6 +679,9 @@ private[kafka] abstract class Acceptor(val socketServer: SocketServer,
   def close(): Unit = {
     beginShutdown()
     thread.join()
+    if (!started) {
```

Comment:
Do we want to put the `started` as a parameter to the method to make it look cleaner?

Reply:
Hey @ex172000 thanks for the review! Do you mean as a parameter to closeAll?

I don't think that's necessary as started is an instance variable. Also, I didn't add the condition to finally block because it should always be true at that point in the code, so it's only present on this code path where the thread has not been started yet.

Reply:
(Similar concern with `Processor::start` / `Processor::close`)